bigchaindb/backend/mongodb/query.py

```diff
@@ -83,6 +85,43 @@ def get_blocks_status_from_transaction(conn, transaction_id):
 
 
 @register_query(MongoDBConnection)
+def get_txids_filtered(conn, asset_id, operation=None):
+    parts = []
+
+    if operation != Transaction.TRANSFER:
+        # get the txid of the create transaction for asset_id
+        cursor = conn.db['bigchain'].aggregate([
+            {'$match': {
+                'block.transactions.id': asset_id,
+                'block.transactions.operation': 'CREATE'
+            }},
+            {'$unwind': '$block.transactions'},
+            {'$match': {
+                'block.transactions.id': asset_id,
+                'block.transactions.operation': 'CREATE'
+            }},
+            {'$project': {'block.transactions.id': True}}
+        ])
+        parts.append(elem['block']['transactions']['id'] for elem in cursor)
+
+    if operation != Transaction.CREATE:
```

Comment:
Well it would have to be `if operation is None or operation == Transaction.TRANSFER`